MACHINE-LEARNING-SERVICE/src/controllers/objectRecognition.controller.js

```diff
@@ -16,44 +16,34 @@ const Decompress = require('../helpers/decompress.helper');
 const { threadId } = require('worker_threads');
 const CocoSsd = require('../models/cocoSsd.model');
 const Yolo = require('../models/yolo.model');
+const InvalidFileException = require('../Exceptions/invalidFile.exception');
+const HandlerModel = require('../models/handler.model');
 
 //Controls the model that will be used to detect the object
 class ObjectRecognitionController {
 
   //Returns the results of the detection according to the model, object and percentage indicated
   static async recognizeObject(req, res) {
     const { zipName, percentage, object, model } = req.body;
-    const decompressedFilePath = Decompress.decompressFile(
-      `${__dirname}/../uploads/zips/${zipName}`
-    );
-    if (!decompressedFilePath) {
-      res.send('The file has not been unziped');
-      return;
-    }
-    if (model == 'coco') {
-      const cocoSsd = new CocoSsd(
-        path.join(__dirname, '../uploads/images/'),
-        percentage,
-        object
-      );
-      const result = await cocoSsd.predict();
-      if (result.length === 0) {
-        res.send(`There is not the object ${object} in the image`);
-      }
-      res.send({ result });
-    } else if (model == 'yolo') {
-      const yolo = new Yolo(
-        path.join(__dirname, '../uploads/images/'),
-        percentage,
-        object
+    console.log(typeof zipName !== 'undefined');
+    
+    try{
+      const decompressedFilePath = Decompress.decompressFile(
+        `${__dirname}/../uploads/zips/${zipName}`
       );
-      const result = await yolo.predict();
-      if (result.length === 0) {
-        res.send(`There is not the object ${object} in the image`);
+      if (!decompressedFilePath) {
+        res.send('The file has not been unziped');
+        return;
```

Comment:
You could do this in order to have one line: 
return res.send('The file has not been unziped');